event version

metacoma opened the pull request.

metacoma committed:
event version

omgbebebe commented on event_processing.txt:
```diff
@@ -6,30 +6,23 @@ JetStreamChannel: 'KN_USER_{{ USER }}__{{ HOST }}_BROKER_KNE_TRIGGER' #example:
 Forward cloudevents from ContextBroker in to HostBroker
     - any:
-        - cesql: source LIKE 'org.mindwm.graph.%' # example org.mindwm.graph.node or org.mindwm.graph.node.relationship
+        - cesql: source LIKE 'org.mindwm.graph.{{ context_name }}.%' # example org.mindwm.graph.pink.node or org.mindwm.graph.pink.node.relationship
```
IMO should be `org.mindwm.graph.pink.relationship` instead of `org.mindwm.graph.pink.node.relationship`.
```
org.mindwm.graph.pink.node
org.mindwm.graph.pink.relationship
```

omgbebebe replied:
Why do you put a `{{ context_name }}` to after the object? In my mind at the first we should list all sources and then the subject. I.e.
```
org -- where
mindwm -- where
pink -- where
graph -- object
node -- object
```
Thus, `where` is `org.mindwm.pink`, and `what` is `graph.node`. What wrong with that approach?

metacoma replied:
The main problem is that we pass user data (context name) to the third 'semantic level,' which is also used for the knfunc and manager naming conventions

For example, what if the user specifies the context name as `manager` or `knfunc`? 
In this case:
```
org.mindwm.knfunc.graph.node
org.mindwm.manager.graph.relationship
```
Overlaps with another convention
```
org.mindwm.knfunc.{{ func_name }}
org.mindwm.manager.{{ user_name }}
```

omgbebebe replied:
make sense. Let try to name the source more precisely. I.e.
```
org.mindwm.manager.{{ user }}.{{ host }}.< tmux | terminal | clipboard | etc >
org.mindwm.core.{{ context_name }}.graph.node
org.mindwm.core.{{ context_name }}.graph.relationship
org.mindwm.core.{{ context_name }}.knfunc.llm_answer
org.mindwm.core.{{ context_name }}.knfunc.show_message
org.mindwm.zulip.{{ org_name }}.{{ stream_name }}
```
maybe will be better if we merge `username` with `hostname` as `{{ username }}@{{ hostname }}`, then it took only one field in the `source`. Something like:
```
org.mindwm.manager.{{ user }}@{{ host }}.< tmux | terminal | clipboard | etc >
org.mindwm.manager.pion@stg1.tmux.{{ base64 }}.{{ session }}.{{ pane }}
```
then we get
```
org.mindwm -- constant prefix
<core | manager | extension > -- source of event
< user@host | context_name | zulip > -- event agent
< rest > -- source in terms of an agent
```

If we change `core` name to `context` maybe it will make more sense:
```
org.mindwm.manager.{{ user }}.{{ host }}.< tmux | terminal | clipboard | etc >
org.mindwm.context.{{ context_name }}.graph.node
org.mindwm.context.{{ context_name }}.graph.relationship
org.mindwm.context.{{ context_name }}.knfunc.llm_answer
org.mindwm.context.{{ context_name }}.knfunc.show_message
org.mindwm.zulip.{{ org_name }}.{{ stream_name }}
```

WDYT?

metacoma replied:
LGTM

metacoma replied:
> maybe will be better if we merge username with hostname as {{ username }}@{{ hostname }}, then it took only one field in the source. 

I'm okay with it, but it seems like it will complicate nats filters and cesql expressions, such as:
```
cesql: "source LIKE 'org.mindwm.manager.user@%.tmux.1'"
```
instead of
```
cesql: "source LIKE 'org.mindwm.manager.user.%.tmux.1'"
```
Also, in the case of NATS, this will remove one 'level' in the NATS topic hierarchy

```
org.mindwm.manager.{{ user }}.{{ host }}.tmux # 5 levels
org.mindwm.manager.{{user }}@{{ host }}.tmux # 4 leveles
```

And we also need to design and implement an additional 'addressing' `{{ user }}@{{ host }}` system throughout the entire EDA.

omgbebebe replied:
> cesql: "source LIKE 'org.mindwm.manager.user@%.tmux.1'"

Why do you need to get `each first tmux session on every host of this user`? For such weird thing we will get weird filters for sure.:) I don't mind to keep old naming if you wish.

> org.mindwm.manager.{{user }}@{{ host }}.tmux # 4 leveles

In my mind this is just a tag `some user at some host`. If we need to filter all events from particular user for some reason, then we shuld keep old dot-separated naming convention. I'm ok with that.

metacoma committed:
fix typo

metacoma committed:
refact

subject: org.mindwm.manager.{{ user }}.{{ host }}.< tmux | terminal | clipboard | etc >
source: org.mindwm.context.{{ context_name }}.graph.node
source: org.mindwm.context.{{ context_name }}.graph.relationship
source: org.mindwm.context.{{ context_name }}.knfunc.llm_answer
source: org.mindwm.context.{{ context_name }}.knfunc.show_message
source: org.mindwm.zulip.{{ org_name }}.{{ stream_name }}

omgbebebe reviewed (approved):
LGTM
